src/base58.cpp

```diff
@@ -31,11 +41,11 @@ bool DecodeBase58(const char* psz, std::vector<unsigned char>& vch)
     // Process the characters.
     while (*psz && !isspace(*psz)) {
         // Decode base58 character
-        const char* ch = strchr(pszBase58, *psz);
-        if (ch == nullptr)
+        if ((uint8_t)*psz >= sizeof(mapBase58)/sizeof(mapBase58[0]))
```

Comment:
Just make the mapBase58 array 256 bytes large, and you don't need the range check.